### PR TITLE
updated toPredicate and all existing specifications to use the context

### PR DESCRIPTION
**Pull request by marijnritense**

### CODE REVIEW

File: case/src/main/kotlin/com/ritense/case/web/rest/CaseTabResource.kt [hunk 1]
```diff
@@ -42,4 +44,13 @@ class CaseTabResource(
         return ResponseEntity.ok(tabs)
     }
 
+    @GetMapping("/v1/document/{documentId}/tab")
```
Comment (theo-ritense) [inline case/src/main/kotlin/com/ritense/case/web/rest/CaseTabResource.kt:48]: We should probably deprecate the old one

File: case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt [hunk 2]
```diff
@@ -47,8 +55,9 @@ class CaseTabResourceIntTest : BaseIntegrationTest() {
     @WithMockUser(username = "user@ritense.com", authorities = [USER])
     fun `should get case tabs`() {
         val caseDefinitionName = "some-case-type"
+        val document = createDocument(caseDefinitionName)
         mockMvc.perform(
-            get("/api/v1/case-definition/{caseDefinitionName}/tab", caseDefinitionName)
+            get("/api/v1/document/{documentId}/tab", document.id.id)
```
Comment (theo-ritense) [inline case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt:94]: For as long as we don't remove the old endpoint yet, we should keep the tests for it.

File: case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt [hunk 3]
```diff
@@ -66,8 +75,9 @@ class CaseTabResourceIntTest : BaseIntegrationTest() {
     @WithMockUser(username = "user@ritense.com", authorities = [ADMIN])
     fun `should get case tabs filtered for role`() {
         val caseDefinitionName = "some-case-type"
+        val document = createDocument(caseDefinitionName)
         mockMvc.perform(
-            get("/api/v1/case-definition/{caseDefinitionName}/tab", caseDefinitionName)
+            get("/api/v1/document/{documentId}/tab", document.id.id)
```
Comment (theo-ritense) [inline case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt:114]: For as long as we don't remove the old endpoint yet, we should keep the tests for it.

File: authorization/src/main/kotlin/com/ritense/authorization/permission/Permission.kt [hunk 4]
```diff
@@ -141,14 +170,14 @@ data class Permission(
...
-                        resourceType,
+                        request.resourceType,
                         queryDialectHelper
                     )
                 }.toTypedArray()
             )
     }
 
-    private fun <U> appliesInContext(
+    fun <U> appliesInContext(
```
Comment (theo-ritense) [inline authorization/src/main/kotlin/com/ritense/authorization/permission/Permission.kt:180]: This method is short, but hard to understand. Can you break the boolean checks up?

File: case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt (hunk 2 above)
Comment (marijnritense) [reply case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt:94]: added back in

File: case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt (hunk 3 above)
Comment (marijnritense) [reply case/src/test/kotlin/com/ritense/case/web/rest/CaseTabResourceIntTest.kt:114]: added back in

File: case/src/main/kotlin/com/ritense/case/web/rest/CaseTabResource.kt (hunk 1 above)
Comment (marijnritense) [reply case/src/main/kotlin/com/ritense/case/web/rest/CaseTabResource.kt:48]: Done

File: authorization/src/main/kotlin/com/ritense/authorization/permission/Permission.kt (hunk 4 above)
Comment (marijnritense) [reply authorization/src/main/kotlin/com/ritense/authorization/permission/Permission.kt:180]: I don't see how it is hard to understand, it's an if statement with parenthesis. If you want to make it more readable, I'm assuming you have something in mind. Can you make the change yourself?

Comment (theo-ritense) [reply authorization/src/main/kotlin/com/ritense/authorization/permission/Permission.kt:180]: done

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**